### PR TITLE
Fix regex pattern in jinja macros for sudo related rules.

### LINKED ISSUE #6076: Ansible playbook for e8 profile fails on RHEL8
#### Description of problem:
When applying Ansible playbook for e8 profile it fails with the following error:
```
TASK [Remove lines containing NOPASSWD from sudoers files] *********************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: sre_constants.error: bad escape \N at position 14
failed: [192.168.122.121] (item={'path': '/etc/sudoers'}) => {"ansible_loop_var": "item", "changed": false, "item": {"path": "/etc/sudoers"}, "module_stderr": "Shared connection to 192.168.122.121 closed.\r\n", "module_stdout": "Traceback (most recent call last):\r\n  File \"/root/.ansible/tmp/ansible-tmp-1599950583.7995758-48836-108024660234250/AnsiballZ_replace.py\", line 102, in <module>\r\n    _ansiballz_main()\r\n  File \"/root/.ansible/tmp/ansible-tmp-1599950583.7995758-48836-108024660234250/AnsiballZ_replace.py\", line 94, in _ansiballz_main\r\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\r\n  File \"/root/.ansible/tmp/ansible-tmp-1599950583.7995758-48836-108024660234250/AnsiballZ_replace.py\", line 40, in invoke_module\r\n    runpy.run_module(mod_name='ansible.modules.files.replace', init_globals=None, run_name='__main__', alter_sys=True)\r\n  File \"/usr/lib64/python3.6/runpy.py\", line 205, in run_module\r\n    return _run_module_code(code, init_globals, run_name, mod_spec)\r\n  File \"/usr/lib64/python3.6/runpy.py\", line 96, in _run_module_code\r\n    mod_name, mod_spec, pkg_name, script_name)\r\n  File \"/usr/lib64/python3.6/runpy.py\", line 85, in _run_code\r\n    exec(code, run_globals)\r\n  File \"/tmp/ansible_replace_payload_x4gg4u9u/ansible_replace_payload.zip/ansible/modules/files/replace.py\", line 302, in <module>\r\n  File \"/tmp/ansible_replace_payload_x4gg4u9u/ansible_replace_payload.zip/ansible/modules/files/replace.py\", line 271, in main\r\n  File \"/usr/lib64/python3.6/re.py\", line 233, in compile\r\n    return _compile(pattern, flags)\r\n  File \"/usr/lib64/python3.6/re.py\", line 301, in _compile\r\n    p = sre_compile.compile(pattern, flags)\r\n  File \"/usr/lib64/python3.6/sre_compile.py\", line 562, in compile\r\n    p = sre_parse.parse(p, flags)\r\n  File \"/usr/lib64/python3.6/sre_parse.py\", line 855, in parse\r\n    p = _parse_sub(source, pattern, flags & SRE_FLAG_VERBOSE, 0)\r\n  File \"/usr/lib64/python3.6/sre_parse.py\", line 416, in _parse_sub\r\n    not nested and not items))\r\n  File \"/usr/lib64/python3.6/sre_parse.py\", line 765, in _parse\r\n    p = _parse_sub(source, state, sub_verbose, nested + 1)\r\n  File \"/usr/lib64/python3.6/sre_parse.py\", line 416, in _parse_sub\r\n    not nested and not items))\r\n  File \"/usr/lib64/python3.6/sre_parse.py\", line 502, in _parse\r\n    code = _escape(source, this, state)\r\n  File \"/usr/lib64/python3.6/sre_parse.py\", line 401, in _escape\r\n    raise source.error(\"bad escape %s\" % escape, len(escape))\r\nsre_constants.error: bad escape \\N at position 14\r\n", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}

PLAY RECAP *********************************************************************
192.168.122.121            : ok=306  changed=75   unreachable=0    failed=1    skipped=28   rescued=0    ignored=0
```

#### SCAP Security Guide Version:
master

#### Operating System Version:
RHEL8

### PR DESCRIPTION
**Pull request by ggbecker**
#### Description:

- Fix jinja macros for sudo related rules.

- Fixes #6076 

- Supersedes #6081

### CODE REVIEW

**Comment by mildas** [comment]
**Changes identified:**
Rule sudo_remove_no_authenticate:
 Found change in bash remediation.
 Template usage changed in ansible remediation.
 Ansible remediation changed.
Rule sudo_require_authentication:
 Found change in bash remediation.
 Template usage changed in ansible remediation.
 Ansible remediation changed.
Rule sudo_remove_nopasswd:
 Found change in bash remediation.
 Ansible remediation changed.
Macro ansible_sudo_remove_config:
 In Ansible remediation for sudo_require_authentication.
 In Ansible remediation for sudo_remove_no_authenticate.
 In Ansible remediation for sudo_remove_nopasswd.
Macro bash_sudo_remove_config:
 In Bash remediation for sudo_remove_nopasswd.
 In Bash remediation for sudo_remove_no_authenticate.
 In Bash remediation for sudo_require_authentication.

**Recommended tests to execute:**
 build_product ol8
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol8-ds.xml sudo_require_authentication
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol8-ds.xml sudo_require_authentication
 build_product debian10
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-debian10-ds.xml sudo_remove_no_authenticate
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-debian10-ds.xml sudo_remove_no_authenticate
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-debian10-ds.xml sudo_remove_nopasswd
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-debian10-ds.xml sudo_remove_nopasswd

File: linux_os/guide/system/software/sudo/sudo_remove_no_authenticate/ansible/shared.yml [hunk 1]
```diff
@@ -4,4 +4,4 @@
 # complexity = low
 # disruption = low
 
-{{{ ansible_sudo_remove_config("!authenticate", "!authenticate") }}}
+{{{ ansible_sudo_remove_config("!authenticate", "\!authenticate") }}}
```
Comment (mildas) [inline linux_os/guide/system/software/sudo/sudo_remove_no_authenticate/ansible/shared.yml:7]: Is the backslash needed? The remediation works for me even without it.

Comment (ggbecker) [reply linux_os/guide/system/software/sudo/sudo_remove_no_authenticate/ansible/shared.yml:7]: It should work both ways... I was just following the same regex from existent OVAL:

https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/software/sudo/sudo_remove_no_authenticate/oval/shared.xml#L37

**Comment by ggbecker** [comment]
@mildas can you take a look at this pull request again? Thanks

**Comment by mildas** [comment]
Sure, I'm on it.

**Comment by mildas** [comment]
Tests are passing and the changes look good to me. Thank you!

**Pull request merged**